server/utils/taskUtils.ts

```diff
@@ -1,9 +1,90 @@
 import { Task } from '../@types/shared'
+import { TableCell, TableRow } from '../@types/ui'
+import { DateFormats } from './dateUtils'
+import { sentenceCase } from './utils'
 
-type GroupedTask = { allocated: Array<Task>; unallocated: Array<Task> }
+const DUE_DATE_APPROACHING_DAYS_WINDOW = 3
+
+const allocatedTableRows = (tasks: Array<Task>): Array<TableRow> => {
```

Comment:
I wonder if this new code should be in a `tasks` subfolder, then we can have all the table-related code in a file of it's own (something like `table.ts`), then we can import it in an `index.ts`, so we can use the entire module all in one. Just keeps things a bit tidier is all.